src/Aspire.Dashboard/wwwroot/js/app.js

```diff
@@ -435,3 +435,20 @@ window.focusElement = function(selector) {
         element.focus();
     }
 }
+
+window.getWindowDimensions = function() {
+    return {
+        width: window.innerWidth,
+        height: window.innerHeight
+    }
+}
+
+window.listenToWindowResize = function(dotnetHelper) {
+    window.addEventListener('load', () => {
+        dotnetHelper.invokeMethodAsync('OnResizeAsync', { width: window.innerWidth, height: window.innerHeight });
+    });
+
+    window.addEventListener('resize', () => {
+        dotnetHelper.invokeMethodAsync('OnResizeAsync', { width: window.innerWidth, height: window.innerHeight });
```

Comment:
The resize event fires very frequently when the browser is being resized with a mouse. We don't want to raise hundreds of events. This should be debounced.

For example, https://stackoverflow.com/a/52256801

Reply:
I added a 100ms debounce time

Reply:
I tried this out and I think this needs to be tweaked. Debouncing delays the event until resize is finished. That means you keep the old view until resize is finished. Not a great experience.

Instead, throttling the event would be better: https://hackernoon.com/optimizing-performance-with-throttling-in-javascript. The browser can tell the server it is resizing every 150ms.

Reply:
This is much better, thanks. Changed to throttling